src/ngraph/distributed.cpp

```diff
@@ -14,37 +14,90 @@
 // limitations under the License.
 //*****************************************************************************
 
-#ifdef NGRAPH_DISTRIBUTED
+#ifdef NGRAPH_DISTRIBUTED_ENABLE
```

Comment:
Consider splitting this into multiple files - one for each type of transport (i.e., MLSL vs. MPI)